File: M4 Intermediate DSA/src/L14_Sorting/Assignment2.java
package L14_Sorting;

import java.util.Arrays;

public class Assignment2 {


  public int solve(int[] A) {
    int N = A.length;

    Arrays.sort(A);

    if (A[N - 1] == 0) {
      return 1;
    }

    int count = 0;

    for (int index = N - 2; index >= 0; index--) {
      if (A[index] != A[index + 1]) {
        count = N - index - 1;
      }

      if (A[index] == count) {
        return 1;
      }


    }
    return -1;
  }
}

File: M4_Intermediate_DSA/L3_Sorting/assignment2.go
package main

import "sort"

func SolveA2(A []int) int {
	count := 0

	sort.Slice(A, func(i, j int) bool {
		return A[i] > A[j]
	})

	if A[0] == 0 {
		return 1
	}

	for i := 1; i < len(A); i++ {
		if A[i] != A[i-1] {
			count = i
		}
		if A[i] == count {
			return 1
		}
	}

	return -1
}
